test/sagemaker_tests/huggingface_pytorch/training/integration/sagemaker/test_smdp.py

```diff
@@ -75,6 +74,9 @@ def test_smdp_question_answering(ecr_image, instance_type, py_version, sagemaker
     """
     Tests SM Distributed DataParallel single-node via script mode
     """
+    transformers_version = ecr_image.split('transformers')[1][:5]
```

Comment:
@mansimane Could you please use regular expressions to find the transformers version?

The less likely way for this to fail is if "transformers` were to be mentioned elsewhere in the image URI string, which would result in the string parse being incorrect.

The more likely way for this to fail is when the version has any double-digit numbers in the major/minor/patch version. So `4.10.2` would be parsed incorrectly because it is 6 characters long.

```suggestion
    transformers_version_search = re.search(r"transformers(\d+(\.\d+){1,2})")
    if transformers_version_search:
        transformers_version = transformers_version_search.group(1)
    else:
        raise LookupError("HF transformers version not found in image URI")
```
or something similar.

Reply:
Good catch, added commit with the fix.